express-post-json/server.js

```diff
@@ -0,0 +1,27 @@
+import express from 'express';
+
+const app = express();
+app.use(express.json());
+
+let nextId = 1;
+const grades = {};
+
+app.get('/api/grades', (req, res) => {
+  const json = [];
+  for (const key in grades) {
+    json.push(grades[key]);
+  }
+  res.json(json);
+});
+
+app.post('/api/grades', (req, res) => {
+  const newEntry = req.body;
+  newEntry.id = nextId;
+  grades[nextId] = newEntry;
+  nextId++;
+  res.status(201).json(newEntry);
```

Comment:
You should check for errors, like an id that doesn't exist. The way this code is, if they give you a non-existent ID you will create a new entry. That's not the spec.